stripe/src/main/java/com/stripe/android/FingerprintDataRepository.kt

```diff
@@ -1,24 +1,43 @@
 package com.stripe.android
 
 import android.content.Context
+import androidx.annotation.UiThread
+import androidx.annotation.WorkerThread
 import com.stripe.android.networking.FingerprintRequestExecutor
 import com.stripe.android.networking.FingerprintRequestFactory
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 import java.util.Calendar
 import kotlin.coroutines.CoroutineContext
 
 internal interface FingerprintDataRepository {
+    @UiThread
```

Comment:
should this be `@WorkerThread` ?

Reply:
`@UiThread` because it is safe to call from the main thread.

Reply:
hmm from the [doc](https://developer.android.com/reference/kotlin/androidx/annotation/UiThread) a method is annotated with `@UiThread` only when it _has_ to be run on ui thread - but it seems `#refresh` doesn't have to be called on ui thread right?